site/en/docs/extensions/reference/declarativeNetRequest/index.md

```diff
@@ -158,102 +150,145 @@ examples of URL filters:
+An extension can specify up to **50** static [rulesets](#type-Ruleset) as part of the `"rule_resources"` manifest key, but only **10** of these rulesets can be enabled at a time. The latter is called the [`MAX_NUMBER_OF_ENABLED_STATIC_RULESETS`](#property-MAX_NUMBER_OF_ENABLED_STATIC_RULESETS). Collectively, those rulesets are guaranteed at least **30,000** rules. This is called the [`GUARANTEED_MINIMUM_STATIC_RULES`](#property-GUARANTEED_MINIMUM_STATIC_RULES). The number of rules available after that depends on how many rules are enabled by all the extensions installed on a user's browser. You can find this number at runtime by calling [`getAvailableStaticRuleCount()`](#method-getAvailableStaticRuleCount). You can see [an example of this](#update-static-rulesets) under [code examples](#code-examples).
 
-When rules are applied to browsers with pages in the service worker's cached storage, the browser may ignore the set rule for those specific pages until the cached storage is cleared. This is because cached storage is intended to be persistent, and many features like offline use do not expect the cache to be cleared without also clearing a service worker's registration as well. For cases when extensions utilizing declarativeNetRequest must be enabled and disabled repeatedly, the [`chrome.browsingData`](/docs/extensions/reference/browsingData/) API may be used to clear the cache to guarantee proper functionality.
+#update-static-rulesets
 
-## Example {: #example }
+#### Dynamic and session rules {: #dynamic-session-rules }
 
-{% Label %}manifest.json:{% endLabel %}
+The limits applied to dynamic and session rules are simpler than static rules. The total number of both cannot exceed **5000**. This is called the [`MAX_NUMBER_OF_DYNAMIC_AND_SESSION_RULES`][#property-MAX_NUMBER_OF_DYNAMIC_AND_SESSION_RULES].
 
-```json
-{
-  "name": "declarativeNetRequest extension",
-  "version": "1",
-  "declarative_net_request": {
-    "rule_resources": [{
-      "id": "ruleset_1",
-      "enabled": true,
-      "path": "rules.json"
-    }]
-  },
-  "permissions": [
-    "declarativeNetRequest"
-  ],
-  "host_permissions": [
-    "*://*.google.com/*",
-    "*://*.abcd.com/*",
-    "*://*.example.com/*",
-    "https://*.xyz.com/*",
-    "*://*.headers.com/*",
-  ],
-  "manifest_version": 3
+#### Rules that use regex {: #regex-rules }
+
+All types of rules can use regular expressions; however the total number of regex rules, of each type cannot exceed **1000**. This is called the [MAX_NUMBER_OF_REGEX_RULES](#property-MAX_NUMBER_OF_REGEX_RULES).
+
+Additionally, each rule must be less than 2KB once compiled. This roughly correlates with the complexity of the rule. If you try to load a rule that exceeds this limit, you will see a warning like the one below and the rule will be ignored.
+
+```bash
+rules_1.json: Rule with id 1 specified a more complext regex than allowed as part of the "regexFilter" key.
```

Comment:
Can we make the `pre` wrap the content here? Otherwise maybe add linebreaks.

Reply:
Done.